.github/workflows/ci.yml

```diff
@@ -44,6 +44,12 @@ jobs:
         run: |
           cd frontend
           npm run check
+      - name: Run tests
```

Comment:
Could you please add run tests before running lints? Usually you want to get the most valuable signal first, to avoid a situation where CI fails on formatting/lints before it tells you that tests are actually broken.